### PR TITLE
Fix timezone for VZ DAGs & clean up DAGs overall 🧹 

### LINKED ISSUE #12780: Investigate / Improve Airflow V2 logging of exceptions
A possible downside to the containerization of DAGs means that exceptions thrown inside the docker container that is doing the task may be emitted on STDERR or otherwise are not captured by the XCOM logging system. This can obscure the actual problem at hand and require the operator to simulate the execution of the task inside the production airflow environment. This issue intends to track the discussion and work to remediate this disadvantage, possibly through better exception handling via the docker library inside the DAG or some manipulation of the STDOUT and STDERR data streams to merge these into the STDOUT which is logged "normally" during a successful run.

## background

As seen in the airflow log for a failed DAG execution:

```
9b28c28974d8
*** Found local files:
***   * /opt/airflow/logs/dag_id=vz-cris-import/run_id=manual__2023-07-10T12:23:36.529683+00:00/task_id=invoke_cris_import_via_docker/attempt=1.log
[2023-07-10, 07:23:37 CDT] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: vz-cris-import.invoke_cris_import_via_docker manual__2023-07-10T12:23:36.529683+00:00 [queued]>
[2023-07-10, 07:23:37 CDT] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: vz-cris-import.invoke_cris_import_via_docker manual__2023-07-10T12:23:36.529683+00:00 [queued]>
[2023-07-10, 07:23:37 CDT] {taskinstance.py:1308} INFO - Starting attempt 1 of 1
[2023-07-10, 07:23:37 CDT] {taskinstance.py:1327} INFO - Executing <Task(_PythonDecoratedOperator): invoke_cris_import_via_docker> on 2023-07-10 12:23:36.529683+00:00
[2023-07-10, 07:23:37 CDT] {standard_task_runner.py:57} INFO - Started process 1439 to run task
[2023-07-10, 07:23:37 CDT] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'vz-cris-import', 'invoke_cris_import_via_docker', 'manual__2023-07-10T12:23:36.529683+00:00', '--job-id', '5135', '--raw', '--subdir', 'DAGS_FOLDER/vz_cris_import.py', '--cfg-path', '/tmp/tmpe0q6_89a']
[2023-07-10, 07:23:37 CDT] {standard_task_runner.py:85} INFO - Job 5135: Subtask invoke_cris_import_via_docker
[2023-07-10, 07:23:37 CDT] {task_command.py:410} INFO - Running <TaskInstance: vz-cris-import.invoke_cris_import_via_docker manual__2023-07-10T12:23:36.529683+00:00 [running]> on host 9b28c28974d8
[2023-07-10, 07:23:37 CDT] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='vz-cris-import' AIRFLOW_CTX_TASK_ID='invoke_cris_import_via_docker' AIRFLOW_CTX_EXECUTION_DATE='2023-07-10T12:23:36.529683+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-07-10T12:23:36.529683+00:00'
[2023-07-10, 07:23:40 CDT] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 220, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/vz_cris_import.py", line 33, in invoke_cris_import_via_docker
    auto_remove=True,
  File "/home/airflow/.local/lib/python3.7/site-packages/docker/models/containers.py", line 888, in run
    container, exit_status, command, image, out
docker.errors.ContainerError: Command 'None' in image 'atddocker/vz-cris-import:latest' returned non-zero exit status 1
[2023-07-10, 07:23:40 CDT] {taskinstance.py:1350} INFO - Marking task as FAILED. dag_id=vz-cris-import, task_id=invoke_cris_import_via_docker, execution_date=20230710T122336, start_date=20230710T122337, end_date=20230710T122340
[2023-07-10, 07:23:40 CDT] {standard_task_runner.py:109} ERROR - Failed to execute job 5135 for task invoke_cris_import_via_docker (Command 'None' in image 'atddocker/vz-cris-import:latest' returned non-zero exit status 1; 1439)
[2023-07-10, 07:23:40 CDT] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-07-10, 07:23:40 CDT] {taskinstance.py:2651} INFO - 0 downstream tasks scheduled from follow-on schedule check
```

VS the equivalent execution from inside a worker container:

```
root@9b28c28974d8:/opt/airflow# docker run -it --rm atddocker/vz-cris-import:latest
Traceback (most recent call last):
  File "/app/cris_import.py", line 821, in <module>
    main()
  File "/app/cris_import.py", line 34, in main
    secrets = get_secrets()
  File "/app/cris_import.py", line 197, in get_secrets
    return onepasswordconnectsdk.load_dict(client, REQUIRED_SECRETS)
  File "/usr/local/lib/python3.10/dist-packages/onepasswordconnectsdk/config.py", line 66, in load_dict
    item_vault = _vault_uuid_for_field(field=field, vault_tag=vault_tag)
  File "/usr/local/lib/python3.10/dist-packages/onepasswordconnectsdk/config.py", line 159, in _vault_uuid_for_field
    raise NoVaultSetForFieldException(
onepasswordconnectsdk.config.NoVaultSetForFieldException: There         is no vault for SFTP_endpoint field
root@9b28c28974d8:/opt/airflow#
```

### PR DESCRIPTION
**Pull request by frankhereford**
## Associated issues

This PR will close https://github.com/cityofaustin/atd-data-tech/issues/12780.

## Associated repo

Vision Zero

## Testing

This work came out of my inadvertent use of UTC as the timezone of choice for the VZ DAGs. While I was in there, a few other opportunities came up to clean up, so this PR does:

* [x] Use central time for VZ DAGs
* [x] Use the DockerOperator instead of the python library
* [x] Get CRIS import running on `production` tag of the image to match VZ CI
* [x] Simplify passing the environment down to the docker containers
* [x] Add slack notifications to AFD/EMS
* [x] Use the `tty` parameter to Docker, 🙏 @johnclary!!

**Steps to test:**

* [ ] Spin up the Vision Zero local stack DB
* [ ] Spin up this branch in local airflow and you should be able to trigger these DAGs.

---
#### Ship list
- [ ] Code reviewed

### CODE REVIEW

File: dags/vz_afd_ems_import.py
Comment (frankhereford) [inline dags/vz_afd_ems_import.py:1]: These are heavily reworked, and almost entirely with the delete key. It's much easier to look at the final file: https://github.com/cityofaustin/atd-airflow/blob/fix-timezone/dags/vz_afd_ems_import.py

File: dags/vz_cris_import.py
Comment (frankhereford) [inline dags/vz_cris_import.py:1]: Same here. Easier to review final file: https://github.com/cityofaustin/atd-airflow/blob/fix-timezone/dags/vz_cris_import.py

**Review by chiaberry — APPROVED** [review]
a lot of refactoring in a short time! ✨

File: dags/vz_afd_ems_import.py [hunk 1]
```diff
@@ -1,82 +1,57 @@
...
-# EMS DAG
-
-# define the parameters of the DAG
+# EMS
 @dag(
     dag_id="vz-ems-import",
     description="A DAG which imports EMS data into the Vision Zero database.",
     schedule="0 7 * * *",
-    start_date=pendulum.datetime(2023, 1, 1, tz="UTC"),
+    start_date=pendulum.datetime(2023, 1, 1),
```
Comment (mddilley) [inline dags/vz_afd_ems_import.py:13]: TIL that pendulum inherits datetime so it ~is~ supports all of its methods 👍

File: dags/vz_afd_ems_import.py [hunk 2]
```diff
@@ -1,82 +1,57 @@
...
 import os
 import pendulum
-from airflow.decorators import dag, task
+from airflow.decorators import dag
+from airflow.operators.docker_operator import DockerOperator
+from utils.slack_operator import task_fail_slack_alert
 
-DEPLOYMENT_ENVIRONMENT = os.environ.get("ENVIRONMENT", 'development')   # our current environment from ['production', 'development']
 
-ENVIRONMENT = {
```
Comment (mddilley) [inline dags/vz_afd_ems_import.py:8]: This caught my eye at first, but I see why this is removed since it is happening again in the script that is run. 👍

**Review by mddilley — APPROVED** [review]
Thanks for updating these. 🙏 I tested the DAGs locally, and I saw everything green. ✅ 🚀

File: dags/vz_afd_ems_import.py (hunk 2 above)
Comment (frankhereford) [reply dags/vz_afd_ems_import.py:8]: Thanks for observing that! The airflow environment has the 3 1PW related keys and the deployment class (production or development), and the scripts look for those values in the same environment variable names, so I can just pass the whole environment down into the container with `environment=dict(os.environ),` in the `DockerOperator`.

**Comment by frankhereford** [comment]
Follow up on this and why there will probably be another micro-PR to clean up TZ a little more: [here](https://austininnovation.slack.com/archives/CHZE6BC6L/p1689202206829259?thread_ts=1689199106.688049&cid=CHZE6BC6L).

> Hm, this is pesky.
> We have a situation where our local macs have their clocks set to local time with no offset because local time is local time. On atd-data03, we have our clock set to UTC with an a designation to the computer to show times in America/Chicago because we run the server in the central timezone. This is an example of Macs bucking UNIX trends because.. apple, I guess.
> My PR opted to not set a timezone because not specifying means use the system clock. This looked perfect on my local stack because that runs on top of my mac, but on atd-data03, the interval is being interpreted in UTC which makes the schedule wrong.
> I’m going to think on this. Either we specify in our DAGs that we are talking Americas/Chicago or we set atd-data03 to have its clock in local time. I suspect route one is more correct.

**Comment by frankhereford** [comment]
image

**Pull request merged**